docs/cudf/source/user_guide/memory-profiling.md

```diff
@@ -2,12 +2,15 @@
 
 # Memory Profiling
 
-
-Peak memory usage is a common concern in GPU programming since the available GPU memory is typically less than available CPU memory. To easily identify memory hotspots, cudf provides a memory profiler.
+Peak memory usage is a common concern in GPU programming since the available GPU memory is typically less than available CPU memory. To easily identify memory hotspots, cudf provides a memory profiler. In comes with an overhead so avoid using it in performance-sensitive code.
 
 ## Enabling memory profiling
 
-1. Use `cudf.set_option`:
+First, we need to enable memory profiling in [RMM](https://docs.rapids.ai/api/rmm/stable/guide/). One way to do this is by calling {py:func}`rmm.statistics.enable_statistics()`. This will add a statistics resource adaptor to the current RMM memory resource, which enables cudf to access memory profiling information. See the RMM documentation for more details.
+
+Second, we need to enable memory profiling in cudf by using either of the following:
```

Comment:
This is confusing because you have "First..." and "Second", but also "One way to do this..." in between these. Then you have two numbered bullets. So there are too many numbered things in this section. Can you simplify it? 

Even better, why do we have to enable memory profiling in RMM at all -- can't cuDF handle that for us if we enable it in cuDF?

Reply:
Tried to simplify it a bit. 

> Even better, why do we have to enable memory profiling in RMM at all -- can't cuDF handle that for us if we enable it in cuDF?

I did this initially but as @wence- points out, by enabling RMM statistics implicitly: [This doesn't behave like a normal context manager (leave the state unchanged after exit)](https://github.com/rapidsai/cudf/pull/15866#discussion_r1650602241)

Reply:
OK got it.